src/Common/TransformEndianness.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <base/Decimal_fwd.h>
+#include <base/extended_types.h>
+
+#include <utility>
+
+namespace DB
+{
+template <std::endian endian, typename T>
+    requires is_big_int_v<T>
+inline void transformEndianness(T & x)
+{
+    if constexpr (std::endian::native != endian)
+    {
+        std::ranges::transform(x.items, std::begin(x.items), [](auto& item) { return std::byteswap(item); });
+        std::ranges::reverse(x.items);
+    }
+}
+
+template <std::endian endian, typename T>
+    requires is_decimal<T> || std::is_floating_point_v<T>
+inline void transformEndianness(T & value)
```

Comment:
I don't like `Decimal` and floating point values being in one handler. `Decimal`s are essentially integers. We should transform them the same way as other integers (l. 37) - with `std::byteswap(value)`. Let's move them into a separate case.

Reply:
True, thanks!